.github/workflows/docker-publish.yml

```diff
@@ -42,17 +52,40 @@ jobs:
     needs: cache-deps
     runs-on: ubuntu-latest
     permissions:
-      contents: read
+      id-token: write
+      contents: write
       packages: write
 
     steps:
+      - name: Audit DNS requests
+        if: github.repository == 'cds-snc/cra-arc-upd-tbpc'
+        uses: cds-snc/dns-proxy-action@main
+        env:
+          DNS_PROXY_FORWARDTOSENTINEL: 'true'
+          DNS_PROXY_LOGANALYTICSWORKSPACEID: ${{ secrets.LOG_ANALYTICS_WORKSPACE_ID }}
+          DNS_PROXY_LOGANALYTICSSHAREDKEY: ${{ secrets.LOG_ANALYTICS_WORKSPACE_KEY }}
+
       - name: Checkout repository
         uses: actions/checkout@v4
 
+      - name: Configure aws credentials using OIDC
+        if: github.repository == 'cds-snc/cra-arc-upd-tbpc'
+        uses: aws-actions/configure-aws-credentials@b47578312673ae6fa5b5096b330d9fbac3d116df # v4.2.1
+        with:
+          role-to-assume: arn:aws:iam::211125499457:role/cra-arc-upd-tbpc-apply
+          role-session-name: ECRPush
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Login to Amazon ECR
+        if: github.repository == 'cds-snc/cra-arc-upd-tbpc'
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@062b18b96a7aff071d4dc91bc00c4c1a7945b076 # v2.0.1
+
       - name: Log in to the Container registry
+        if: github.repository != 'cds-snc/cra-arc-upd-tbpc'
```

Comment:
Do you need to log into to the container registry? Can you build the image locally? In general, we use something like that to push to the ECR - https://github.com/cds-snc/url-shortener/blob/main/.github/workflows/build_and_push_staging.yml#L34
You build it locally, then login and push to the ECR.